### ISSUE TITLE
ObjectContainer breaks when position is minus infinity

### ISSUE THREAD

**Issue by insidiator** [melonjs/melonJS#444]
When I set position to -Infinity and width to Infinity there are multiple problems that prevent a container from being drawn.
-  `camera.isVisible` will always return false because `right` and `bottom` are NaN (`-Infinity+Infinity`). https://github.com/melonjs/melonJS/blob/1.0.0-dev/src/shapes/rectangle.js#L136 and https://github.com/melonjs/melonJS/blob/1.0.0-dev/src/shapes/rectangle.js#L154
  - this could be fixed by checking if the width/height are Infinity. In that case the position should be ignored.
- `globalTranslation` is translated by `obj.pos` and afterwards moved in the reverse direction which again produces NaN and breaks more stuff as `globalTranslation` won't be reset to a valid number. https://github.com/melonjs/melonJS/blob/1.0.0-dev/src/renderable/container.js#L723
  - saving the `globalTranslation` position at the beginning of the `update` method and resetting it to that value would fix this.

```
            var isTranslated;
            var x = globalTranslation.pos.x;
            var y = globalTranslation.pos.y;
            var viewport = me.game.viewport;
...

                    if (isTranslated) {
                        globalTranslation.pos.set(x, y);
                    }
```
- everything inside the container gets translated by `-Infinity` and thus won't show on the screen. https://github.com/melonjs/melonJS/blob/1.0.0-dev/src/renderable/container.js#L761
  - This could be avoided if the container translates its children respecting the anchor point. In cases where any of the bounds are Infinite, the anchor point should be at 0 in that axis in order to prevent children from being moved to Infinity.

The first two points should not pose a problem, but the third one will break pretty much everything, unless `ObjectContainer` sets the `anchorPoint` to [0,0]. 
If you are okay with the proposed solution, I can create a pull request.

**Comment by parasyte** [comment]
For infinite planes, we need to rework our entire architecture; everything needs to be relative to an arbitrary point of origin, instead of upper-left-corner.

**Comment by obiot** [comment]
with the latest version of Tiled supporting Infinite world, that would be nice to have this one working properly !

**Comment by FW4248** [comment]
> When I set position to -Infinity and width to Infinity there are multiple problems that prevent a container from being drawn.
> 
> * `camera.isVisible` will always return false because `right` and `bottom` are NaN (`-Infinity+Infinity`). https://github.com/melonjs/melonJS/blob/1.0.0-dev/src/shapes/rectangle.js#L136 and https://github.com/melonjs/melonJS/blob/1.0.0-dev/src/shapes/rectangle.js#L154
> * this could be fixed by checking if the width/height are Infinity. In that case the position should be ignored.
> * `globalTranslation` is translated by `obj.pos` and afterwards moved in the reverse direction which again produces NaN and breaks more stuff as `globalTranslation` won't be reset to a valid number. https://github.com/melonjs/melonJS/blob/1.0.0-dev/src/renderable/container.js#L723
>   
>   * saving the `globalTranslation` position at the beginning of the `update` method and resetting it to that value would fix this.
> 
> ```
>             var isTranslated;
>             var x = globalTranslation.pos.x;
>             var y = globalTranslation.pos.y;
>             var viewport = me.game.viewport;
> ...
> 
>                     if (isTranslated) {
>                         globalTranslation.pos.set(x, y);
>                     }
> ```
> 
> * everything inside the container gets translated by `-Infinity` and thus won't show on the screen. https://github.com/melonjs/melonJS/blob/1.0.0-dev/src/renderable/container.js#L761
>   
>   * This could be avoided if the container translates its children respecting the anchor point. In cases where any of the bounds are Infinite, the anchor point should be at 0 in that axis in order to prevent children from being moved to Infinity.
> 
> The first two points should not pose a problem, but the third one will break pretty much everything, unless `ObjectContainer` sets the `anchorPoint` to [0,0]. If you are okay with the proposed solution, I can create a pull request.
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/n8n-io/n8n?pullRequest=3437) <br/>All committers have signed the CLA.
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/n8n-io/n8n?pullRequest=3437) <br/>All committers have signed the CLA.
_Originally posted by @CLAassistant in https://github.com/n8n-io/n8n/issues/3437#issuecomment-> [![CLA assistant check](https://camo.githubusercontent.com/4c970efeed75dd01f6778bf7e33dec85e4171375cb69acb0927bfc9d437d78d9/68747470733a2f2f636c612d617373697374616e742e696f2f70756c6c2f62616467652f6e6f745f7369676e6564)](https://cla-assistant.io/n8n-io/n8n?pullRequest=3437) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/n8n-io/n8n?pullRequest=3437) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/n8n-io/n8n?pullRequest=3437) it.
https://winksite.com/site/resources/banner-image-template.png

**Comment by FW4248** [comment]
[Uploading export-FW4248-1654966774.csv…]()

**Comment by FW4248** [comment]
> 

-![ **[** ]( Uploading export-FW4248-1654966774.csv…)

**Comment by obiot** [comment]
closing this one, see #1122
